### PR TITLE
Refactor j2me localization support

### PR DESCRIPTION
**Pull request by marco-c**
This PR:
1) Fixes a bug (there was a mismatch between en-US.xml and ResourceConstants.java)
2) Stores the localized strings in a JSON file (instead of an XML one)
3) Adds support for languages other than en-US

In a follow-up, I'll add some documentation for localization in the README file.

### CODE REVIEW

**Comment by marco-c** [comment]
(The it-IT.xml file isn't actually localized, I've just translated a couple of strings in order to test loading strings in a different language)

**Comment by marco-c** [comment]
The failure isn't related to this branch.

File: l10n/Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+LANG_FILES=$(shell find . -name "*.xml")
+LANG_DESTS=../java/l10n/$(LANG_FILES:.xml=.json)
+
+$(LANG_DESTS): $(LANG_FILES)
```
Comment (mykmelez) [inline l10n/Makefile]: There's no need to put this rule into a separate Makefile, and [Recursive Make Considered Harmful](http://aegis.sourceforge.net/auug97.pdf), so I would put this into the top-level Makefile, even though it takes files from l10n/ as input and generates output in java/.

File: l10n/xml_to_json.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
```
Comment (mykmelez) [inline l10n/xml_to_json.py]: Nit: I'd put this in tools/.

**Comment by mykmelez** [comment]
Just needs a rebase.

**Pull request merged**